include/wx/msw/ole/activex.h

```diff
@@ -68,76 +69,48 @@ class FrameSite;
 //
 //---------------------------------------------------------------------------
 
-template<typename I>
-class wxAutoOleInterface
+template <class T>
+class wxAutoOleInterface : public wxCOMPtr<T>
```

Comment:
Inheriting from a smart pointer is rather weird. It works here because it's COM, but from purely C++ point of view I'd very much prefer to use aggregation and have a `wxCOMPtr<T> m_ptr` instead and forward the relevant methods to it. Even if it's more typing, I think this would be more clear and potentially less problematic in the future.